index.js

```diff
@@ -410,7 +410,10 @@ exports.parseNoPatch = function (code, options) {
       ast = jsParse(code, opts);
     }
   } catch (err) {
-    if (err instanceof SyntaxError) {
+    // Any error with location information should be promoted to a code frame.
+    // Otherwise linter plugins for IDEs lose the positional information for
+    // semantic errors.
+    if (err.loc) {
```

Comment:
The only one I can think of is if someone throws an error object with a `loc` property that's not actually a location in a source file. I think that is highly unlikely to begin with, and what's more, since every piece of code being called here is either babylon-lightscript or babel with babel-plugin-lightscript, we have total control of the code path.